### PR TITLE
Implement SQS Proxy Lambda (Provisioning)

### PR DESCRIPTION
**Pull request by juliusfitzhugh-ccpo**
Implement func to consume up to 10 SQS messages when a `GET /provisioning-jobs` request is sent
- Update infrastructure to utilize `consume-provisioning-job` func
- Implement `consume-provisioning-job` func and unit tests
- Move a some provisioning components into `provisioning-sfn-workflow`
- Update infrastructure unit tests
- Add `logger` tool from `lambda-powertools`

Ticket: AT-7058

### CODE REVIEW

File: lib/atat-web-api-stack.ts [hunk 1]
```diff
@@ -55,24 +51,22 @@ export class AtatWebApiStack extends cdk.Stack {
...
-    this.provisioningStateMachine = new StateMachine(this, "ProvisioningStateMachine", {
-      stateMachineProps: {
-        definition: provisioningSfn.workflow,
-      },
-      logGroup: provisioningSfn.logGroup,
-    }).stateMachine;
 
     // Provisioning lambda that translates and invokes the state machine
     const provisioningJob = new ApiSfnFunction(this, "ProvisioningJobRequest", {
       method: HttpMethod.POST,
```
Comment (kylelaker-ccpo) [inline lib/atat-web-api-stack.ts:59]: Have we talked about https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_apigateway-readme.html#aws-stepfunctions-backed-apis?

Comment (kylelaker-ccpo) [reply lib/atat-web-api-stack.ts:59]: ehh maybe more specifically https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_apigateway.StepFunctionsIntegration.html

**Review by kylelaker-ccpo — APPROVED** [review]

File: api/provision/consume-provisioning-job.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+const PROVISIONING_QUEUE_URL = process.env.PROVISIONING_QUEUE_URL ?? "";
+const logger = new Logger({ serviceName: "provisioningWorkflow" });
+
+export async function baseHandler(event: APIGatewayProxyEvent): Promise<APIGatewayProxyResult> {
+  logger.info("General lambda info", { isAColdStart: logger.isColdStart() });
+  // poll messages from the queue
+  const receiveMessageInput: ReceiveMessageCommandInput = {
+    QueueUrl: PROVISIONING_QUEUE_URL,
+    MaxNumberOfMessages: 10,
```
Comment (jasonburkert-ccpo) [inline api/provision/consume-provisioning-job.ts:20]: Has there been discussion of parameterizing this quantity?  This is good for now.

File: lib/atat-web-api-stack.ts (hunk 1 above)
Comment (jasonburkert-ccpo) [reply lib/atat-web-api-stack.ts:59]: I don't recall previous discussions of StepFunctions backed APIs

File: package.json [hunk 3]
```diff
@@ -38,16 +38,17 @@
...
     "@aws-sdk/client-sfn": "^3.58.0",
     "@aws-sdk/client-sqs": "^3.58.0",
     "@middy/core": "^2.5.7",
     "@middy/http-json-body-parser": "^2.5.7",
     "@middy/validator": "^2.5.7",
     "aws-cdk-lib": "2.17.0",
+    "cdk-iam-floyd": "0.345.0",
     "cdk-nag": "^2.12.5",
     "constructs": "^10.0.98",
-    "cdk-iam-floyd": "0.345.0",
```
Comment (jasonburkert-ccpo) [inline package.json:50]: Should we bother to remove these changes since it's now in develop?

File: utils/middleware/error-handling-middleware.ts [hunk 4]
```diff
@@ -11,9 +11,15 @@ import {
...
 
-export type MiddlewareInputs = StepFunctionRequestEvent<RequestBodyType> | CspInvocation | ProvisionRequest;
+export type MiddlewareInputs =
+  | StepFunctionRequestEvent<RequestBodyType>
+  | CspInvocation
+  | ProvisionRequest
+  | APIGatewayProxyEvent;
 export type MiddlewareOutputs = APIGatewayProxyResult | CspResponse | ValidationErrorResponse | ProvisionRequest;
 
+// A central place to catch an handle errors that occur before,
```
Comment (jasonburkert-ccpo) [inline utils/middleware/error-handling-middleware.ts:21]:
```suggestion
// A central place to catch and handle errors that occur before,
```

File: api/provision/consume-provisioning-job.test.ts [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    extendedRequestId: undefined,
+    cfId: undefined,
+    attempts: 1,
+    totalRetryDelay: 0,
+  },
+  Messages: [
+    {
+      MessageId: "b5353c",
+      ReceiptHandle: "AQC0e6b=",
+      MD5OfBody: "b03",
```
Comment (jasonburkert-ccpo) [inline api/provision/consume-provisioning-job.test.ts:22]: Did you have to calculate these MD5s?  Or just update value after test failure?

File: api/provision/consume-provisioning-job.test.ts [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+    expect(response.statusCode).toBe(SuccessStatusCode.OK);
+    expect(responseBody).toHaveLength(0);
+    expect(responseBody).toEqual([]);
+  });
+});
+
+describe("Consumer Provisioning Job invalid behavior", () => {
+  it("should throw an internal error if invalid response from queue", async () => {
+    process.env.PROVISIONING_QUEUE_URL = "my url";
+    sqsMock.on(ReceiveMessageCommand).resolves(undefined!);
```
[the remark this answers is not in the document]
Comment (juliusfitzhugh-ccpo) [reply api/provision/consume-provisioning-job.test.ts:90]: The intent here was trying to mimic if some unknown response caused the function to fail an error response would still be sent with `internal error` and  I ended up putting `undefined!`.

File: package.json (hunk 3 above)
Comment (kylelaker-ccpo) [reply package.json:50]: I think this is an artifact of the merge conflict on the file

File: lib/atat-web-api-stack.ts (hunk 1 above)
Comment (juliusfitzhugh-ccpo) [reply lib/atat-web-api-stack.ts:59]: Yeah it was [mentioned previously](https://github.com/dod-ccpo/atat-web-api/pull/638#discussion_r826276454), and I thought a ticket was supposed to come out of it, so I didn't think about changing it.

File: package.json (hunk 3 above)
Comment (juliusfitzhugh-ccpo) [reply package.json:50]: I'll already rebased, so it should remove that.

File: api/provision/consume-provisioning-job.test.ts (hunk 5 above)
Comment (juliusfitzhugh-ccpo) [reply api/provision/consume-provisioning-job.test.ts:22]: No I didn't calculate the MD5s, but got them from previous function executions. I shorted it down so it doesn't look like a secret and cause a false positive.

File: api/provision/consume-provisioning-job.ts (hunk 2 above)
Comment (juliusfitzhugh-ccpo) [reply api/provision/consume-provisioning-job.ts:20]: No I don't think so.

File: lib/atat-web-api-stack.ts (hunk 1 above)
Comment (jasonburkert-ccpo) [reply lib/atat-web-api-stack.ts:59]: Thank you for the reminder.  I didn't connect the two.  I do have a note to create that ticket from #638, but haven't yet done so.

File: package.json (hunk 3 above)
Comment (juliusfitzhugh-ccpo) [reply package.json]: Yeah, appears to be an artifact like Kyle mentioned.

File: lib/atat-web-api-stack.ts (hunk 1 above)
Comment (kylelaker-ccpo) [reply lib/atat-web-api-stack.ts:59]: Oh it's AT-7177!

**Pull request merged**